AttributeError: 'NoneType' object has no attribute 'get_extraction_methods'

With certain versions of Python and/or specific and targeted scorn from God, when starting ComfyUI with this node pack you get a mess of nonsense about extraction methods and a bunch of other gibberish, for example:


[ERROR] An error occurred while retrieving information for the 'AudioFeatureExtractor' node.
Traceback (most recent call last):
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\tooltips\tooltip_manager.py", line 173, in input_types_with_tooltips
    input_types = original_input_types.get(cls, cls)()
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\nodes\flex\feature_extractors.py", line 22, in INPUT_TYPES
    extraction_methods = feature_class.get_extraction_methods()
AttributeError: 'NoneType' object has no attribute 'get_extraction_methods'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\tooltips\tooltip_manager.py", line 173, in input_types_with_tooltips
    input_types = original_input_types.get(cls, cls)()
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\nodes\flex\feature_extractors_audio.py", line 12, in INPUT_TYPES
    parent_inputs = super().INPUT_TYPES()["required"]
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\tooltips\tooltip_manager.py", line 177, in input_types_with_tooltips
    return original_input_types.get(cls, cls)()
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\nodes\flex\feature_extractors.py", line 22, in INPUT_TYPES
    extraction_methods = feature_class.get_extraction_methods()
AttributeError: 'NoneType' object has no attribute 'get_extraction_methods'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\tooltips\tooltip_manager.py", line 173, in input_types_with_tooltips
    input_types = original_input_types.get(cls, cls)()
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\nodes\flex\feature_extractors.py", line 22, in INPUT_TYPES
    extraction_methods = feature_class.get_extraction_methods()
AttributeError: 'NoneType' object has no attribute 'get_extraction_methods'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\ComfyUI\server.py", line 591, in get_object_info
    out[x] = node_info(x)
  File "D:\ComfyUI\server.py", line 558, in node_info
    info['input'] = obj_class.INPUT_TYPES()
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\tooltips\tooltip_manager.py", line 177, in input_types_with_tooltips
    return original_input_types.get(cls, cls)()
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\nodes\flex\feature_extractors_audio.py", line 12, in INPUT_TYPES
    parent_inputs = super().INPUT_TYPES()["required"]
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\tooltips\tooltip_manager.py", line 177, in input_types_with_tooltips
    return original_input_types.get(cls, cls)()
  File "D:\ComfyUI\custom_nodes\comfyui_ryanontheinside\nodes\flex\feature_extractors.py", line 22, in INPUT_TYPES
    extraction_methods = feature_class.get_extraction_methods()
AttributeError: 'NoneType' object has no attribute 'get_extraction_methods'

--- a/__init__.py
+++ b/__init__.py
@@ -1,7 +1,10 @@
-from .tooltips import TooltipManager, apply_tooltips
-from .tooltips.categories import register_all_tooltips
+# Set this to True to disable all tooltips
+DISABLE_TOOLTIPS = False
 
-# Register tooltips immediately after import
+if not DISABLE_TOOLTIPS:
+    from .tooltips import TooltipManager, apply_tooltips
+    from .tooltips.categories import register_all_tooltips
+
 from comfy.utils import ProgressBar
 from tqdm import tqdm
 from .node_configs.node_configs import CombinedMeta
@@ -654,4 +657,5 @@ if hasattr(PromptServer, "instance"):
         [web.static("/ryanontheinside_web_async", (Path(__file__).parent.absolute() / "ryanontheinside_web_async").as_posix())]
     )
 #register tooltips after all classes are initialized
-register_all_tooltips()
+if not DISABLE_TOOLTIPS:
+    register_all_tooltips()

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,7 +1,7 @@
 [project]
 name = "comfyui_ryanonyheinside"
 description = "Custom nodes introducing everything-reactivity and timelines to ComfyUI"
-version = "2.0.5"
+version = "2.0.6"
 license = {file = "LICENSE"}
 
 dependencies = ["pygame","opencv-python==4.10.0","scipy","torchaudio", "pillow","librosa==0.10.2","pymunk==6.8.1","matplotlib","openunmix","mido","scikit-image"]

--- a/tooltips/__init__.py
+++ b/tooltips/__init__.py
@@ -3,9 +3,7 @@ Tooltip management system for ComfyUI nodes.
 """
 
 from .tooltip_manager import TooltipManager, apply_tooltips
-from .categories import register_all_tooltips
 
 __all__ = ['TooltipManager', 'apply_tooltips']
 
-# Register tooltips when the module is imported
-register_all_tooltips()
+

